### ISSUE TITLE
Run skill-v2.1-sys on Qwen3-32B (python-tiny) to disentangle skill-version from baseline strength

### ISSUE THREAD

**Issue by bdravec** [bdravec/merge-conflict-skill#82]
## Motivation

The completed **v1** large-pair run shows the skill (sys condition) **hurts** the strong Qwen3-32B baseline on python-tiny:

| metric | no-skill | skill-v1-sys | Δ |
|---|---|---|---|
| edit-sim | 0.5321 | 0.5065 | **−0.0256** |
| winnowing | 0.6337 | 0.6129 | **−0.0208** |

- Paired n=3475 common scored cases; win/tie/loss = 416/2285/774 (sign-test z≈10, skill degrades ~2× more cases than it improves).
- Every bucket is negative; no subgroup benefits.

This is consistent with the **"skill direction = baseline strength"** finding (skill helped the weaker Apertus 8B baseline at v2.1-sys: +7.10pp; hurts the stronger Qwen3-32B here).

## Confound to resolve

The 8B win was on **skill v2.1**, but the 32B run above used **skill v1**. So the −2.5pp at 32B could be partly a *skill-version* effect rather than purely *baseline strength*. A **v2.1-sys** run on Qwen3-32B uses the same skill version as the 8B win, isolating the baseline-strength axis.

## Task

- [ ] Run `pilot.py --model qwen3-32b --skill-version v2.1` (sys condition only, per large-pair convention) on python-tiny, all buckets, against the Qwen3-32B vLLM serve on the RTX 6000 box.
- [ ] Reuse the existing no-skill baseline (`pilot_results_qwen3-32b_baseline_python_tiny_32b.jsonl`, 3577 cases) for the paired comparison.
- [ ] Report skill-v2.1-sys vs no-skill (edit/winnowing/exact, per-bucket, win/tie/loss) and compare the v2.1-vs-v1 delta on 32B.

## Notes
- Large-pair runs are **sys-only** (no no-skill/user in the skill run).
- Watch `text+sytx+func` (long-prompt bucket) for the #80 vLLM-hang pattern; `REQUEST_TIMEOUT=300` is in place.
- Refs #75.

**Comment by bdravec** [comment]
## Result — v1/v2/v2.1-sys vs no-skill on Qwen3-32B (python-tiny)

All three skill versions run sys-only on the RTX 6000 (refs #75), paired against the shared no-skill baseline (`pilot_results_qwen3-32b_baseline_python_tiny_32b.jsonl`, 3577 cases). Paired on cases scored in both the version and the baseline.

| version | edit-sim Δ | winnowing Δ | exact-match (skill vs base) | win/tie/loss |
|---|---|---|---|---|
| **v1** | −0.0256 | −0.0208 | 3.31% vs 4.09% ❌ | 416 / 2285 / 774 |
| **v2** | −0.0145 | −0.0172 | 4.44% vs 4.04% ✅ | 535 / 2301 / 630 |
| **v2.1** | **−0.0031** | **−0.0049** | **5.12% vs 4.06%** ✅ | 582 / 2315 / **578** |

Cross-version head-to-head (edit-sim, paired) is monotone: **v2.1 > v2 > v1** (v2.1 beats v1 by +0.0225, v2 by +0.0117).

### Verdict (resolves the v1-vs-v2.1 confound)
The −2.5pp the skill cost at **v1** was **mostly skill-version, not baseline strength.** With the mature skill:
- **Edit/winnowing reach parity** with no-skill (−0.003 / −0.005, within noise; win/tie/loss a dead heat at 582/578).
- **Exact-match goes net-positive**: v2.1 produces **178 perfect resolutions vs the baseline's 141** (5.12% vs 4.06%, +26% relative) — and exact-match is the gold metric for merge resolution.
- Per-bucket, v2.1 is flat-to-positive on most buckets (func ≈0, sytx+func +0.0002, text+func +0.0032, text+sytx+func +0.0044); residual negatives only in `text` (−0.013) and `text+sytx` (−0.026, n=80, noisy).

### Refined RQ3 framing
A *mature* skill is **neutral-to-positive even on a strong baseline** (Qwen3-32B) and **strongly positive on a weak one** (Apertus-8B v2.1-sys: +7.10pp solved, #66). The skill's marginal benefit scales with baseline weakness, but it is no longer *harmful* at the strong end — refining the earlier "skill direction = baseline strength" finding, which was partly a v1 artifact.

### Reproduce
```
python3 scripts/analyze_skill_versions.py \
  --baseline scripts/results/pilot_results_qwen3-32b_baseline_python_tiny_32b.jsonl \
  --version v1=scripts/results/qwen3-32b_v1_sysonly_clean.jsonl \
  --version v2=scripts/results/qwen3-32b_v2_python_tiny_sysonly_RAW.jsonl \
  --version v2.1=scripts/results/qwen3-32b_v2.1_python_tiny_sysonly_RAW.jsonl
```
(model-agnostic; auto-picks the sys skill condition, collapses the `__resume` dedup)

### Caveats
- One model × python-tiny.
- v2.1 edit/winnowing Δ are within noise — this is **parity**, not a similarity *win*.
- The exact-match gain (+37 cases) looks real but a McNemar test would firm up significance.

Note: this run used skill **v2** and **v2.1** (sys-only) in addition to the original v1 — title says v2.1 but the full v1/v2/v2.1 trend is the actual deliverable.

**Comment by bdravec** [comment]
Resolved. The v1-vs-v2.1 confound is answered: most of the v1 skill deficit on Qwen3-32B was skill-version, not baseline strength. By v2.1 the skill reaches parity on edit/winnowing (−0.003 / −0.005, within noise) and is net-positive on exact-match (5.12% vs 4.06% baseline). Full v1/v2/v2.1 result, verdict, and reproduce command above; data + analysis script committed in f8e2485 / 3199883, pushed to main. Follow-ups (not blocking): McNemar test on the +37 exact-match gain; Apertus-70B large-pair sys runs.
